src/LogOstreamInserters.cpp

```diff
@@ -0,0 +1,200 @@
+	ostream << "Index: " << location.index << '\n';
+	ostream << "CGI extension: " << location.cgiExt << '\n';
+	ostream << "CGI path: " << location.cgiPath << '\n';
+	ostream << "Autoindex: " << location.isAutoindex << '\n';
+	ostream << "LimitExcept:\n";
+	ostream << "  Allowed methods:\n";
+	for (int i = 0; i < MethodCount; ++i) {
+		// NOLINTNEXTLINE: The bool array allowedMethods can never be out of bounds, since it has the same size as MethodCount.
+		if (location.limitExcept.allowedMethods[i])
+			ostream << "    " << static_cast<Method>(i) << '\n';
+	}
+	ostream << "  Allow: " << location.limitExcept.allow << '\n';
+	ostream << "  Deny: " << location.limitExcept.deny << '\n';
+	ostream << "Returns:\n";
+	for (std::map<unsigned short, std::string>::const_iterator it = location.returns.begin();
+		 it != location.returns.end(); ++it) {
+		ostream << "  " << it->first << ": " << it->second << '\n';
+	}
+	return ostream;
+}
+
+/**
+ * @brief Overload << operator to append a ServerConfig.
+ *
+ * @param ostream The output stream.
+ * @param serverConfig The ServerConfig object.
+ * @return std::ostream& The output stream.
+ */
+std::ostream& operator<<(std::ostream& ostream, const ServerConfig& serverConfig)
+{
+	ostream << "Server name: " << serverConfig.serverName << '\n';
+	ostream << "Host: " << serverConfig.host << '\n';
+	ostream << "Port: " << serverConfig.port << '\n';
+	ostream << "Max body size: " << serverConfig.maxBodySize << '\n';
+	ostream << "Error pages:\n";
+	for (std::map<unsigned short, std::string>::const_iterator it = serverConfig.errorPage.begin();
+		 it != serverConfig.errorPage.end(); ++it)
+		ostream << "  " << it->first << ": " << it->second << '\n';
+	ostream << "Locations:\n";
+	for (std::vector<Location>::const_iterator it = serverConfig.locations.begin(); it != serverConfig.locations.end();
+		 ++it)
+		ostream << *it;
+	return ostream;
+}
+
+/**
+ * @brief Overload << operator to append a ConfigFile.
+ *
+ * @param ostream The output stream.
+ * @param configFile The ConfigFile object.
+ * @return std::ostream& The output stream.
+ */
+std::ostream& operator<<(std::ostream& ostream, const ConfigFile& configFile)
+{
+	ostream << "Config file:\n";
+	for (std::vector<ServerConfig>::const_iterator it = configFile.serverConfigs.begin();
+		 it != configFile.serverConfigs.end(); ++it)
+		ostream << *it;
+	return ostream;
+}
+
+/**
+ * @brief Overload << operator to a Method.
+ *
+ * Translates the Method enum to a string.
+ * If the Method is MethodCount, it will output "enum MethodCount".
+ * @param ostream The output stream.
+ * @param method The Method enum.
+ * @return std::ostream& The output stream.
+ */
+std::ostream& operator<<(std::ostream& ostream, Method method)
+{
+	switch (method) {
+	case MethodGet:
+		ostream << "GET";
+		break;
+	case MethodPost:
+		ostream << "POST";
+		break;
+	case MethodDelete:
+		ostream << "DELETE";
+		break;
+	case MethodCount:
+		ostream << "enum MethodCount";
+		break;
```

Comment:
What about the default case, when none of the above cases are hit? We should include

```
default:
       ostream << "UNKNOWN METHOD";
```

Reply:
I deliberately did not include a default: clang-tidy fires a warning if not all cases of an enum over a switch are covered. It gets deactivated when there is a default case, since default catches everything else. 

I researched a bit and found this interesting thread:
--> https://stackoverflow.com/questions/2201493/using-default-in-a-switch-statement-when-switching-over-an-enum

The highest voter would throw an exception (I think as default case), others say assert() and always have default.

LLVM says [Don’t use default labels in fully covered switches over enumerations](https://llvm.org/docs/CodingStandards.html#id48)
They mention -Wswitch which will print a warning when not all cases of an enum are covered in a switch.

Another argument is to throw/abort if none of the cases are reached, not as default but after the switch ran through. Argument being that memory corruption could happen, and if it does aborting is the only option since the program behaves abnormal.

Googles C++ Style Guide explicitly makes an exception for switch cases over enums: [If not conditional on an enumerated value, switch statements should always have a default case (in the case of an enumerated value, the compiler will warn you if any values are not handled).](https://google.github.io/styleguide/cppguide.html#Switch_Statements)

I would prefer to throw outside of the switch case and add -Wswitch to our Makefile
Then everytime we switch over an enum we never add an default, if a new case gets added we get an compile error

Reply:
I can get behind leaving the default case away in order to get the warnings whenever an enum is added and not handled in the switch.

But then we need an additional var which is not set when none of the cases are hit, and will then throw an exception after the switch. That is how I understand the intention of the commenter referring to LLVM and Google.

Reply:
I think LLVM and Google are not that strict, they just say let the compiler warn you.

The stricter variant comes from a commenter of the SO thread who says "what if your program has memory issues and a wrong value gets passed?

We could add an `assert(value >= minEnum && value <= maxEnum)` 
But I think this reaches a level of caution which we don't follow anywhere else

Reply:
I would integrate either the assert() or throw an exception.  Our goal is to write clean and safe code, and here we have identified a learning/practice opportunity. That is also the root reason why we follow (most) of the clang tidy checks and put in the effort to make the changes.

Reply:
Added assert() before switch cases on Method and StatusCode